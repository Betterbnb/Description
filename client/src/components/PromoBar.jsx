import React from 'react';

class PromoBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return(
      <h5>PromoBar</h5>
    )
  }
}

export default PromoBar;