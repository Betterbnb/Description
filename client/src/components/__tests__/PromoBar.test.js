import React from 'react';
import { shallow } from 'enzyme';
import PromoBar from '../PromoBar.jsx';

describe('<PromoBar />', () => {
  let wrapper;

  beforeEach(() => {
    const testData = {
      "miniAd": {
        "title": "This home is on people’s minds.",
        "description": "It’s been viewed 500+ times in the past week."
      },
      "highlights": [
        {
          "title": "Hot tub",
          "description": "This is one of few homes in this area that has this feature.",
          "helpful": 0,
          "notHelpful": 0
        },
        {
          "title": "Great value",
          "description": "94% of recent guests gave this home’s value a 5-star rating.",
          "helpful": 0,
          "notHelpful": 0
        }
      ]
    };
    wrapper = shallow(<PromoBar {...testData}/>);
  });

  it('should render correctly with props', () => {
    expect(wrapper).toMatchSnapshot();
  });


  xit('should change helpful thumb icon color on mouseover', () => {
    //write test for changing helpful thumb icon on mouseover
    //https://stackoverflow.com/questions/49934975/how-to-simulate-mouse-over-event-on-a-div-using-enzyme-for-testing-a-react-appli
  });

});