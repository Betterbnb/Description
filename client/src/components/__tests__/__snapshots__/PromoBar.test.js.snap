// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PromoBar /> should render correctly with props 1`] = `
<ForwardRef>
  <ForwardRef>
    <ForwardRef
      src="https://a0.muscache.com/airbnb/static/page3/icon-uc-light-bulb-b34f4ddc543809b3144949c9e8cfcc8d.gif"
    />
    <div
      style={
        Object {
          "fontWeight": "400",
        }
      }
    >
      This home is on people’s minds.
    </div>
    <div>
      It’s been viewed 500+ times in the past week.
    </div>
  </ForwardRef>
  <ForwardRef>
    <ForwardRef>
      HOME HIGHLIGHTS
    </ForwardRef>
    <div
      key="0"
    >
      <div>
        <span
          style={
            Object {
              "fontWeight": "400",
            }
          }
        >
          Hot tub
           · 
        </span>
        This is one of few homes in this area that has this feature.
      </div>
      <ForwardRef>
        <ForwardRef
          href="#"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#898989",
            }
          }
        >
          Helpful
          <ForwardRef
            src="https://s3-us-west-1.amazonaws.com/betterbnb-description/defaultHelpful.png"
          />
        </ForwardRef>
        <span>
          · 
        </span>
        <ForwardRef
          href="#"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#898989",
            }
          }
        >
          Not helpful
        </ForwardRef>
      </ForwardRef>
    </div>
    <div
      key="1"
    >
      <div>
        <span
          style={
            Object {
              "fontWeight": "400",
            }
          }
        >
          Great value
           · 
        </span>
        94% of recent guests gave this home’s value a 5-star rating.
      </div>
      <ForwardRef>
        <ForwardRef
          href="#"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#898989",
            }
          }
        >
          Helpful
          <ForwardRef
            src="https://s3-us-west-1.amazonaws.com/betterbnb-description/defaultHelpful.png"
          />
        </ForwardRef>
        <span>
          · 
        </span>
        <ForwardRef
          href="#"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#898989",
            }
          }
        >
          Not helpful
        </ForwardRef>
      </ForwardRef>
    </div>
  </ForwardRef>
</ForwardRef>
`;
